### PR TITLE
Show 'new' metrics on visitor engagement report/sparklines

### LINKED ISSUE #539: Improve the New Vs Returning report
Currently under Visitors > Frequency and Loyalty, this report is hard to read and analyze.
Instead it should be easier and show more clear New VS Returning metrics.

### PR DESCRIPTION
**Pull request by katebutler**
Fixes #539

### CODE REVIEW

File: plugins/VisitFrequency/Reports/Get.php [hunk 1]
```diff
@@ -32,14 +32,25 @@ protected function init()
         $this->processedMetrics = array(
             new ReturningMetric(new AverageTimeOnSite()),
             new ReturningMetric(new ActionsPerVisit()),
-            new ReturningMetric(new BounceRate())
+            new ReturningMetric(new BounceRate()),
```
Comment (tsteur) [inline plugins/VisitFrequency/Reports/Get.php]: just for readability be good to add `_returning` for the first three returning metrics as a second parameter too then it is more clear. Also I see we use these suffix a few times in the code maybe we could use a constant for it? Also in the `Archive` class where we detect plugin etc (see first change in this diff on github).

File: plugins/VisitFrequency/Reports/Get.php [hunk 2]
```diff
@@ -32,14 +32,25 @@ protected function init()
...
-            'max_actions_returning'
+            'max_actions_returning',
+            'nb_visits_new',
+            'nb_actions_new',
+            'nb_uniq_visitors_new',
+            'nb_users_new',
+            'max_actions_new',
+            'bounce_rate_new',
+            'nb_actions_per_visit_new',
+            'avg_time_on_site_new'
```
Comment (tsteur) [inline plugins/VisitFrequency/Reports/Get.php]: @katebutler I reckon we don't need to add columns like `nb_actions_per_visit_new` etc but only need to add the same columns that are there for returning?

**Review by tsteur — CHANGES_REQUESTED** [review]
@katebutler left a few comments.

File: plugins/VisitFrequency/API.php [hunk 3]
```diff
@@ -36,48 +40,71 @@ class API extends \Piwik\Plugin\API
...
-        $table = Request::processRequest('VisitsSummary.get', $params);
-        $this->prefixColumns($table, $period);
-        return $table;
+        foreach (array_keys($visitTypes) as $columnSuffix) {
+            $this->unprefixColumns($columns, $columnSuffix);
+        }
+
+        /** @var \Piwik\DataTable\Map $resultSet */
+        $resultSet = null;
+
```
[the remark this answers is not in the document]
Comment (katebutler) [reply plugins/VisitFrequency/API.php]: Yup, good idea, have added this

[the remark this answers is not in the document]
Comment (tsteur) [reply plugins/VisitFrequency/API.php]: Can it still happen that visitTypes ends up being empty?

[the remark this answers is not in the document]
Comment (tsteur) [reply plugins/VisitFrequency/API.php]: or could throw an exception if no valid column was passed maybe...

or maybe if no visitTypes exist, then we return all columns?

[the remark this answers is not in the document]
Comment (tsteur) [reply plugins/VisitFrequency/API.php]: Actually, we cannot throw an exception and would just return one set or an empty datatable

**Comment by tsteur** [comment]
@katebutler looking at the tests there seem to be heaps of them failing?

**Comment by tsteur** [comment]
@katebutler can you update the ecommerce tests as discussed last Thursday or Friday? Then we can merge

File: plugins/VisitFrequency/API.php [hunk 4]
```diff
@@ -36,48 +40,80 @@ class API extends \Piwik\Plugin\API
...
+            self::RETURNING_COLUMN_SUFFIX => self::RETURNING_VISITOR_SEGMENT
         );
 
-        $table = Request::processRequest('VisitsSummary.get', $params);
-        $this->prefixColumns($table, $period);
-        return $table;
+        $columns = Piwik::getArrayFromApiParameter($columns);
+
+        /** @var \Piwik\DataTable\DataTableInterface $resultSet */
+        $resultSet = new DataTable();
```
Comment (diosmosis) [inline plugins/VisitFrequency/API.php]: FYI, this needs to be `new Simple()` otherwise the XML output will have an extra `<row>` element.

Comment (tsteur) [reply plugins/VisitFrequency/API.php]: cheers @diosmosis

**Pull request merged**